Snippets/Snippets_5/Features/FeatureConfiguration.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace Snippets5.Features
+{
+    using System;
+    using NServiceBus;
+    using NServiceBus.Features;
+    using NServiceBus.Pipeline;
+
+    class MyMagicFeature : Feature
+    {
+
+#region FeatureConfiguration
+        public MyMagicFeature()
+        {
+            EnableByDefault();
+
+            DependsOn<Sagas>();
+
+            Prerequisite(c => c.Settings.EndpointName().StartsWith("Magical"), "This feature only works on magical endpoints");
+        }
+#endregion
+
+#region FeatureSetup
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            // retrieve some global settings
+            var supervised = context.Settings.Get<bool>("SupervisorActive");
```

Comment:
Wouldn't you still have to check for SupervisorActive = false and therefore end up having 2 if statements?